suisa_sendemeldung/suisa_sendemeldung.py

```diff
@@ -19,328 +18,69 @@
+from typed_settings.exceptions import InvalidValueError
+
+from suisa_sendemeldung.settings import FileFormat, IdentifierMode, OutputMode, Settings
 
 from .acrclient import ACRClient
 
 if TYPE_CHECKING:  # pragma: no cover
-    from argparse import Namespace as ArgparseNamespace
-
     from openpyxl.worksheet.worksheet import Worksheet
 
-_EMAIL_TEMPLATE = """
-Hallo SUISA
-
-Im Anhang finden Sie die Sendemeldung von $station_name für den $month $year.
-
-Diese Sendemeldung erfolgt gemäss den Bestimmungen "Gemeinsamer Tarif S 2020 - 2024" [1] unter
-Einhaltung der Einreichefrist für den Monat $month gemäss Ziffer 45 des Tarifs.
-
-In der Sendungsmeldung enthalten sind die unter Buchstaben G (Verzeichnisse) des Tarifs
-genannten Programmangaben, in elektronischer Form. Die Angaben richten sich nach dem
-standardisierten Format gemäss der "Vorlage für Sendemeldungen: Gemeinsamer Tarif S" [2]
-sowie der Abstimmung zwischen SUISA und $station_name.
-
-
-Diese Sendemeldung enthält unter anderem die unter Ziffer 34 geforderten Angaben:
-
-- Titel des Musikwerks
-- Name des Komponisten
-- Name des bzw. der Hauptinterpreten
-- Label
-- ISRC der benützten Aufnahme (sofern zusammen mit der Aufnahme vom Lieferanten der Aufnahme mitgeliefert)
-- vom Sender der Aufnahme selbst zugewiesene Identifikationsnummer
-- Sendezeit
-- Sendedauer
-
-
-Allfällige Beanstandungen oder technische Rückfragen zu dieser Sendemeldung müssen in
-schriftlicher, elektronischer Form an $responsible_email gerichtet werden.
-
-Beanstandungen müssen, gemäss Ziffer 46, bis spätestens in drei Monaten vor dem $in_three_months
-erfolgen. Anschliessend gilt die Sendemeldung für $month $year als akzeptiert und die
-Tarifbestimmungen als eingehalten.
-
-Korrekt und termingerecht gemeldete Beanstandungen werden unter Berücksichtigung der
-Nachfrist von 45 Tagen, gemäss Ziffer 46, bearbeitet.
-
-Diese Email und Sendemeldung wurde automatisch generiert.
 
-Freundliche Grüsse,
-$station_name
+T = TypeVar("T")
 
-[1] <https://www.suisa.ch/dam/jcr:1d82128b-8cc7-4cf0-b0ef-1890ae9d434d/GTS_2020-2024_GER.pdf>
-[2] <https://www.suisa.ch/dam/jcr:6934be98-4319-4565-8936-05007d83fd35/Vorlage_fuer_Sendemeldungen__GTS__2020-2022.xlsx>
 
---
-$email_footer
-"""  # noqa: E501
-
-_ACRTOKEN_MAXLEN = 32
-
-
-def validate_arguments(parser: ArgumentParser, args: ArgparseNamespace) -> None:
+def validate_arguments(settings: Settings) -> None:
     """Validate the arguments provided to the script.
 
     After this function we are sure that there are no conflicts in the arguments.
 
     Arguments:
     ---------
-        parser: the ArgumentParser to use for throwing errors
-        args: the arguments to validate
+        settings: the Settings instance to validate
+
+    Raises:
+    ------
+        InvalidValueErrors: if there are invalid argument combinations
 
     """
     msgs = []
-    # check length of bearer_token
-    if not len(args.bearer_token) >= _ACRTOKEN_MAXLEN:
```

Comment:
We loose some validation of user provided flags, imo this is fine for our automation use case. I would still prefer to keep the logic so user errors are caught before calling the API.